programs/fuzz/Makefile

```diff
@@ -27,6 +27,10 @@ ifdef FUZZINGENGINE
 LOCAL_LDFLAGS += -lFuzzingEngine
 endif
 
+ifdef WINDOWS_BUILD
```

Comment:
I'm not sure about this, under CMake, the fuzz build is explicitly removed for windows builds, and I am presuming this is why you have not added bcrypt to the cmake libs for fuzz.

Reply:
It has been removed was not needed for Make either.

Reply:
Update. The ci is [failing ](https://mbedtls.trustedfirmware.org/blue/organizations/jenkins/mbed-tls-pr-head/detail/PR-8047-head/19/pipeline) when removing this code

~~~~~~
[2023-09-22T17:23:32.311Z] * build_mingw: build: Windows cross build - mingw64, make (Link Library)
.....
[2023-09-22T17:23:52.954Z]   CC    ssl/ssl_client2.c
[2023-09-22T17:23:53.520Z]  i686-w64-mingw32-gcc common.o onefile.o fuzz_x509csr.o ../../tests/src/helpers.o ../../tests/src/asn1_helpers.o ../../tests/src/psa_crypto_helpers.o ../../tests/src/psa_exercise_key.o ../../tests/src/bignum_helpers.o ../../tests/src/threading_helpers.o ../../tests/src/random.o ../../tests/src/fake_external_rng_for_test.o ../../tests/src/certs.o ../../tests/src/drivers/test_driver_aead.o ../../tests/src/drivers/test_driver_asymmetric_encryption.o ../../tests/src/drivers/test_driver_pake.o ../../tests/src/drivers/test_driver_key_agreement.o ../../tests/src/drivers/test_driver_signature.o ../../tests/src/drivers/test_driver_key_management.o ../../tests/src/drivers/test_driver_cipher.o ../../tests/src/drivers/hash.o ../../tests/src/drivers/test_driver_mac.o ../../tests/src/drivers/platform_builtin_keys.o -L../../library -lmbedtls -lmbedx509 -lmbedcrypto  -o fuzz_x509csr
[2023-09-22T17:23:53.521Z] ../../library/libmbedcrypto.a(entropy_poll.o):entropy_poll.c:(.text+0x35): undefined reference to `BCryptGenRandom@16'
[2023-09-22T17:23:53.521Z] collect2: error: ld returned 1 exit status
[2023-09-22T17:23:53.521Z] Makefile:59: recipe for target 'fuzz_x509csr' failed
[2023-09-22T17:23:53.521Z] make[2]: *** [fuzz_x509csr] Error 1
[2023-09-22T17:23:53.521Z] make[2]: *** Waiting for unfinished jobs....
[2023-09-22T17:23:53.521Z] make[1]: *** [fuzz] Error 2
[2023-09-22T17:23:53.521Z] Makefile:167: recipe for target 'fuzz' failed
[2023-09-22T17:23:53.521Z] Makefile:15: recipe for target 'programs' failed
[2023-09-22T17:23:53.521Z] make: *** [programs] Error 2
[2023-09-22T17:23:53.521Z] ^^^^build_mingw: build: Windows cross build - mingw64, make (Link Library): make CC=i686-w64-mingw32-gcc AR=i686-w64-mingw32-ar LD=i686-w64-minggw32-ld CFLAGS='-Werror -Wall -Wextra' WINDOWS_BUILD=1 lib programs -> 2^^^^
[2023-09-22T17:23:53.779Z] 
[2023-09-22T17:23:53.779Z] ******************************************************************
[2023-09-22T17:23:53.779Z] * Done, cleaning up 
[2023-09-22T17:23:53.779Z] * Fri Sep 22 17:23:53 UTC 2023
[2023-09-22T17:23:53.779Z] ******************************************************************
[2023-09-22T17:23:54.345Z] 
~~~~~~

The test snippet from all.sh is set as:

~~~~~~
component_build_mingw () {
    msg "build: Windows cross build - mingw64, make (Link Library)" # ~ 30s
    scripts/config.py unset MBEDTLS_AESNI_C # AESNI depends on cpu modifiers
    make CC=i686-w64-mingw32-gcc AR=i686-w64-mingw32-ar LD=i686-w64-minggw32-ld CFLAGS='-Werror -Wall -Wextra' WINDOWS_BUILD=1 lib programs

    # note Make tests only builds the tests, but doesn't run them
    make CC=i686-w64-mingw32-gcc AR=i686-w64-mingw32-ar LD=i686-w64-minggw32-ld CFLAGS='-Werror' WINDOWS_BUILD=1 tests
    make WINDOWS_BUILD=1 clean

    msg "build: Windows cross build - mingw64, make (DLL)" # ~ 30s
    make CC=i686-w64-mingw32-gcc AR=i686-w64-mingw32-ar LD=i686-w64-minggw32-ld CFLAGS='-Werror -Wall -Wextra' WINDOWS_BUILD=1 SHARED=1 lib programs
    make CC=i686-w64-mingw32-gcc AR=i686-w64-mingw32-ar LD=i686-w64-minggw32-ld CFLAGS='-Werror -Wall -Wextra' WINDOWS_BUILD=1 SHARED=1 tests
    make WINDOWS_BUILD=1 clean
}
~~~~~~

So linking against the new library is required at least for make. I will be rebasing and removing commit [800223c ](https://github.com/Mbed-TLS/mbedtls/pull/8047/commits/800223c1c42739dfba304eec1b84ef943f4af213). 

We could address the different behaviour between make and cmake in a follow up pr?

Reply:
> We could address the different behaviour between make and cmake in a follow up pr?

The purpose of this PR is to update to later versions of Windows APIs, so let's not make changes to behaviour unless necessary when updating to the later APIs.

(i.e. changing different Make vs CMake behaviour would require design review - specifically, why was it done in the first place? and is that reason still relevant?)